### PR TITLE
stop to discovery addon when its managed cluster is not found

### PR DESCRIPTION
**Pull request by skeeey**
Signed-off-by: Wei Liu <liuweixa@redhat.com>

### CODE REVIEW

File: pkg/hub/addon/discovery_controller.go [hunk 1]
```diff
@@ -128,8 +128,12 @@ func (c *addOnFeatureDiscoveryController) syncAddOn(ctx context.Context, cluster
 	}
 
 	cluster, err := c.clusterLister.Get(clusterName)
+	if errors.IsNotFound(err) {
+		// no cluster, it could be deleted
+		return nil
+	}
 	if err != nil {
-		return fmt.Errorf("unable to find cluster with name %q: %w", clusterName, err)
+		return fmt.Errorf("failed to list cluster with name %q: %w", clusterName, err)
```
Comment (qiujian16) [inline pkg/hub/addon/discovery_controller.go]: why it is list here?

Comment (skeeey) [reply pkg/hub/addon/discovery_controller.go]: roll back the msg, here should be get

**Comment by ldpliu** [comment]
> /cc @ldpliu please take a look at cluster set e2e part, thanks

It looks good to me

**Pull request merged**